src/dso_api/dynamic_api/permissions.py

```diff
@@ -119,70 +119,90 @@ def has_permission_for_models(self, request, view, models):
+        for f in profile.mandatory_filtersets:
+            mandatory.update(f)
 
-            # Everything else is a filter.
-            field_name, op = _parse_filter(key)
+    for key, values in request.GET.lists():
+        if key in ("fields", "format", "page", "page_size", "sorteer") or key.startswith("_"):
+            continue
 
-            # mandatoryFilters may contain either the complete filter (with lookup operation)
-            # or just the field name.
-            if key in mandatory or field_name in mandatory:
-                continue
+        # Everything else is a filter.
+        field_name, op = _parse_filter(key)
 
-            if schema.temporal is not None:
-                if fields := schema.temporal.dimensions.get(field_name):
-                    if any(
-                        not self._filter_ok(field, scopes, schema)
-                        for field in (fields.start, fields.end)
-                    ):
-                        return False
-                    else:
-                        continue
-
-            if not self._filter_ok(field_name, scopes, schema):
-                return False
+        # mandatoryFilters may contain either the complete filter (with lookup operation)
+        # or just the field name.
+        if key in mandatory or field_name in mandatory:
+            continue
 
-        return True
+        if schema.temporal is not None:
+            if fields := schema.temporal.dimensions.get(field_name):
+                if any(
+                    not _filter_ok(field, scopes, schema) for field in (fields.start, fields.end)
+                ):
+                    return False
+                else:
+                    continue
+
+        if not _filter_ok(field_name, scopes, schema):
+            return False
 
-    def _filter_ok(self, field_name: str, scopes: UserScopes, schema: DatasetTableSchema) -> bool:
-        schema_part = schema
-        parts = field_name.split(".")  # Handle subfields.
-        for i, part in enumerate(parts):
-            try:
-                schema_part = schema_part.get_field_by_id(part)
-            except SchemaObjectNotFound as e:
-                # TODO: this should be a ValidationError,
-                # but that exception isn't handled properly from here.
-                # Raise PermissionDenied to at least get the exception message to the client.
-                raise PermissionDenied(f"{field_name} not found in schema") from e
-            if not scopes.has_field_access(schema_part):
+    return True
+
+
+def _filter_ok(  # noqa: C901
+    field_name: str, scopes: UserScopes, schema: DatasetTableSchema
+) -> bool:
+    if field_name == "":
+        raise ValueError("empty field name")
+
+    while field_name != "":
+        try:
+            part, field_name = field_name.split(".", 1)
+        except ValueError:
+            part, field_name = field_name, ""
```

Comment:
This can be done more succinct, without a `try/except`:

```python
part, field_name, *_ = field_name.split(".", 1), ""
```

Reply:
That only works for the no-dot case:

```
>>> field_name = "foo.bar"
>>> part, field_name, *_ = field_name.split(".", 1), ""
>>> part
['foo', 'bar']
>>> field_name
''
```

Reply:
Sorry forgot the `*`

Reply:
As in:
```python
part, field_name, *_ = *field_name.split(".", 1), ""
```

Reply:
And tested:
```python
field_name = "foo.bar"
part, field_name, *_ = *field_name.split(".", 1), ""

print(f"{part=}, {field_name=}")
```
`part='foo', field_name='bar'`

```python
field_name = "fubar"
part, field_name, *_ = *field_name.split(".", 1), ""

print(f"{part=}, {field_name=}")
```
`part='fubar', field_name=''`

Reply:
part, field_name, *_ = *field_name.split(".", 1), ""